src/tools/cgsnapshot.c

```diff
@@ -529,7 +529,12 @@ static int is_ctlr_on_list(char controllers[CG_CONTROLLER_MAX][FILENAME_MAX],
 
 	/* Lets reset the controllers to intersection of controller ∩ wanted_conts */
 	for (i = 0; tmp_controllers[i][0] != '\0'; i++) {
-		snprintf(controllers[i], FILENAME_MAX, "%s", tmp_controllers[i]);
+		int pret;
+
+		pret = snprintf(controllers[i], FILENAME_MAX, "%s", tmp_controllers[i]);
+		if (pret >= FILENAME_MAX)
+			cgroup_err("Warning truncation may have occurred: %s\n",
+				   controllers[i]);
```

Comment:
@drakenclimber Sorry, I had run build against both `gcc-11` and `gcc-10`  but it did not show up because `-Wformat-truncation=` is not enabled by default.  Do you think we could use `memcpy()` here, instead of `snprintf()`  both `Coverity` and `gcc` seems to be happy with it.

Reply:
No worries.

I don't have a strong preference of `memcpy()` vs `snprintf()`.  It would probably be good to add a comment about the truncation warning if we go with something non-standard like `memcpy()`